Readme.md

```diff
@@ -18,171 +18,444 @@ This is an ipm package, which contains one or more reusable assets within the ip
 
 ## Setup
 
-_Add any setup instructions, such as an API Key_
+API key can be generated/accessed by going to this page: https://cloud.google.com/bigquery/docs/authorization
+ 
+Another simple way: Copy and paste the following command 'gcloud auth application-default print-access-token' in Google Cloud Platform shell!
+
 
 ## API<a name="BigQuery"></a>
 
-## BigQuery(auth_key)
+## Functions
+
+<dl>
+<dt><a href="#BigQuery">BigQuery(options)</a> ⇒ <code>Object</code></dt>
+<dd><p>BigQuery object requires an API Key</p>
+</dd>
+<dt><a href="#ExampleDeleteDataset">ExampleDeleteDataset(req, resp)</a></dt>
+<dd><p>Deletes the current dataset which is being referenced</p>
+</dd>
+<dt><a href="#ExampleGetDataset">ExampleGetDataset(req, resp)</a></dt>
+<dd><p>Fetches the given dataset in a project</p>
+</dd>
+<dt><a href="#ExampleInsertAllRows">ExampleInsertAllRows(req, resp)</a></dt>
+<dd><p>Inserts all the rows in the given table</p>
+</dd>
+<dt><a href="#ExampleInsertDataset">ExampleInsertDataset(req, resp)</a></dt>
+<dd><p>Creates a new Dataset in the current project!</p>
+</dd>
+<dt><a href="#ExampleListAllDataset">ExampleListAllDataset(req, resp)</a></dt>
+<dd><p>List All datasets in a project</p>
+</dd>
+<dt><a href="#ExampleUpdateDataset">ExampleUpdateDataset(req, resp)</a></dt>
+<dd><p>Updates i.e. Replaces the current dataset with the passed dataset</p>
+</dd>
+</dl>
+
+<a name="BigQuery"></a>
+
+## BigQuery(options) ⇒ <code>Object</code>
 BigQuery object requires an API Key
 
 **Kind**: global function  
+**Returns**: <code>Object</code> - Project  
 
 | Param | Type | Description |
 | --- | --- | --- |
-| auth_key | <code>string</code> | is the API key for Google BigQuery Service, https://cloud.google.com/bigquery/docs/authorization |
-
+| options | <code>Object</code> | {      authToken: YOUR_BIGQUERY_AUTH_TOKEN (is the API key for Google BigQuery Service, https://cloud.google.com/bigquery/docs/authorization)      (optional parameters)      ,projectID: YOUR_PROJECT_ID      ,datasetID: DATASET_IN_ABOVE_PROJECT      ,tableID: TABLE_IN_ABOVE_DATASET  } |
 
-* [BigQuery(auth_key)](#BigQuery)
-    * [.delete(projectID, datasetID, callback)](#BigQuery+delete)
-    * [.get(projectID, datasetID, callback)](#BigQuery+get)
-    * [.insert(projectID, requestBody, callback)](#BigQuery+insert)
-    * [.insertAll(projectID, datasetID, tableID, requestBody, callback)](#BigQuery+insertAll)
-    * [.list(projectID, callback)](#BigQuery+list)
-    * [.update(projectID, datasetID, requestBody, callback)](#BigQuery+update)
+**Example**  
+```js
+:- Inside of a ClearBlade service
 
-<a name="BigQuery+delete"></a>
+var project = BigQuery(options).initialize();
 
-### bigQuery.delete(projectID, datasetID, callback)
-Datasets - Delete
+Tip: If user wants to work with 
```

Comment:
code snippets in @example should syntactically correct, this line is not

Reply:
Done